src/goose/utils/_cost_calculator.py

```diff
@@ -27,15 +42,20 @@ def _calculate_cost(model: str, token_usage: Usage) -> Optional[float]:
     return None
 
 
-def get_total_cost_message(token_usages: dict[str, Usage]) -> str:
+def get_total_cost_message(
+    token_usages: dict[str, Usage], session_name: str, start_time: datetime, end_time: datetime
+) -> str:
     total_cost = 0
     message = ""
+    session_name_prefix = f"Session name: {session_name}"
     for model, token_usage in token_usages.items():
         cost = _calculate_cost(model, token_usage)
         if cost is not None:
-            message += f"Cost for model {model} {str(token_usage)}: ${cost:.2f}\n"
+            message += f"{session_name_prefix} | Cost for model {model} {str(token_usage)}: ${cost:.2f}\n"
             total_cost += cost
         else:
-            message += f"Cost for model {model} {str(token_usage)}: Not available\n"
-    message += f"Total cost: ${total_cost:.2f}"
-    return message
+            message += f"{session_name_prefix} | Cost for model {model} {str(token_usage)}: Not available\n"
+
+    time_duration = f"{formatted_time(start_time)} - {formatted_time(end_time)}"
```

Comment:
:nit: this is more of datetime range rather than _duration_. we should definitely add time duration though! so it'd look something like

"2024-10-29 00:00:00 - 2024-10-29 00:00: (session duration: 1839 minutes)"  - you could get fancier with "2 hours, 18 minutes, and 22 seconds" but keeping it with a single time unit makes it easier to do analysis on logs (e.g. aggregating total cost and pivoting over time)

Reply:
I like the "time period | session_name | Total cost" format 👍 ! Will change it!

But for duration, I am not sure whether it is useful for the end users at the moment. It is useful for analysis, however, with the raw time, people can calculate the duration with their own preferences such as minutes, seconds.... Definitely we can add this in if the end user asks for it.

Thanks for the review!

Reply:
you can't calculate raw time without parsing the logs, this saves the end user from doing that manually.

to do this the user will have to 
 1. load the file
 2. parse the log file line by line
 3. extract the date time strings to a datetime object,
 4. then convert that to timedelta to do the duration

why burden the end user with all of the unnecessary work if we can write it along side with the log - it's basically free.

Reply:
Hi @lamchau Maybe I missed the request/thread from our users that they would like to see the duration for the total cost.  would you please provide the thread/request?

Reply:
> Maybe I missed the request/thread from our users that they would like to see the duration for the total cost. would you please provide the thread/request?

we indexed on one user requesting session to be tied to logs, i'm also a user that does log analysis on cost. 

> I am just concerning that the line of the cost log is too long if we include too many information. If the user would like to have, we can add in any time

forcing the user to add this in later is much more inconvenient after the fact. long log lines isn't really an issue, the rest of our applications logs are longer (e.g. a serialized `Message` already wraps)